### ISSUE TITLE
Support the Wayland session

### ISSUE THREAD

**Issue by wsdfhjxc** [wsdfhjxc/virtual-desktop-bar#38]
This is a looooooong-term goal. I hope to get it sorted out before 2030.

**Comment by danilacasito** [comment]
yeah, Xorg is too old, supporting it can make this a better project

**Comment by Pan00Pernicek** [comment]
yeah

**Comment by j-maynard** [comment]
This needs to be classified as a bug.  If you have virtual desktop bar in a panel and start a wayland session it causes the wayland session to crash completely.  See plasma bug https://bugs.kde.org/show_bug.cgi?id=442535

**Comment by SamDc73** [comment]
any solution so far ?

**Comment by fauzie811** [comment]
What exactly is preventing it from working on Wayland?

**Comment by SamDc73** [comment]
> What exactly is preventing it from working on Wayland?

when i logged in using in my KDE wayland the Desktop crashed, then it turns out it is because this extention

**Comment by davidedmundson** [comment]
At a minimum can you guard the X calls so that you don't crash the application please even if it doesn't work.
